### PR TITLE
TOOL-47 Add Pact contracts to Analytics React app

### PR DESCRIPTION
**Pull request by irikuo**
[JIRA Ticket TOOL-47](https://reevoo.atlassian.net/browse/TOOL-47)

This PR adds the existing Pact contracts from the old Analytics app. It modifies one of them (the one to get the workbooks) because this app is going to use a new method.

- An env variable has been renamed to be consistent with the rest.
- Some Pact scripts have been added: setup, config and teardown. 
- We have now a more complex set of npm scripts to run each Pact task independently and combined.
- We added some tests to validate the contract data shape.
- The karma config file has been split to handle unit and contract tests independently.

### CODE REVIEW

**Comment by reevoo-samuel** [comment]
*If you're new, read the [guidelines]
(https://github.com/reevoo/guidelines/blob/master/pull_requests.md).*

Good code is **tested**, **designed for change**, **easy to understand**
and **able to cope when things go wrong**.

Good pull requests are **descriptive**, **small** and **short-lived**.

**For the Author**
- [ ] Description explains what you have done and why.
- [ ] Tests have been written (or you have explained why you don't need
tests).
- [ ] Documentation/comments have been added/updated.

**For the Reviewer**
- [ ] The code is well-designed and able to cope with change.
- [ ] The code is easy to understand.
- [ ] Tests cover happy and unhappy paths in the code.
- [ ] Tests are green on CI (or pulled/tested on your machine).

File: README.md [hunk 1]
```diff
@@ -31,6 +31,38 @@ npm start
...
+## Contract
+
+To generate the Pact contract files and upload them to you local Pact Broker, you have to run
+
+```bash
+npm run test:contract
+```
+
+and be sure to have [Pact Broker](https://github.com/reevoo/pact_broker) running.
+
```
Comment (ionut998) [inline README.md]: We now need to run ```bundle install``` as part of the test setup.
Can we add that in the here as well?

**Comment by ionut998** [comment]
There is a lot of noise in the logs when running ```npm test``` is there a way to filter the unnecessary data from here?

```> client_portal-analytics_frontend@0.0.0 pact:teardown /Users/alexandru/reevoo/client_portal-analytics_frontend
> npm run pact:teardown:cp_analytics ; npm run pact:teardown:tableau_gateway


> client_portal-analytics_frontend@0.0.0 pact:teardown:cp_analytics /Users/alexandru/reevoo/client_portal-analytics_frontend
> node tests/pact/setup/teardown.js client_portal_analytics_backend

Pact verified and written.
Upload pact file(s) finished.

> client_portal-analytics_frontend@0.0.0 pact:teardown:tableau_gateway /Users/alexandru/reevoo/client_portal-analytics_frontend
> node tests/pact/setup/teardown.js tableau_api_gateway

Pact verified and written.
Upload pact file(s) finished.

> client_portal-analytics_frontend@0.0.0 pact:stop /Users/alexandru/reevoo/client_portal-analytics_frontend
> npm run pact:stop:cp_analytics ; npm run pact:stop:tableau_gateway


> client_portal-analytics_frontend@0.0.0 pact:stop:cp_analytics /Users/alexandru/reevoo/client_portal-analytics_frontend
> bundle exec pact-mock-service stop -p $npm_package_config_pact_port_cp_analytics_api

[2016-08-15 13:24:28] INFO  going to shutdown ...
[2016-08-15 13:24:28] INFO  WEBrick::HTTPServer#start done.

> client_portal-analytics_frontend@0.0.0 pact:stop:tableau_gateway /Users/alexandru/reevoo/client_portal-analytics_frontend
> bundle exec pact-mock-service stop -p $npm_package_config_pact_port_tableau_gateway_api

[2016-08-15 13:24:29] INFO  going to shutdown ...
[2016-08-15 13:24:29] INFO  WEBrick::HTTPServer#start done.```

**Comment by ionut998** [comment]
LGTM on green

**Comment by irikuo** [comment]
There is a lot of noise because a lot of npm tasks are running, and simplifying the output is not trivial, so it will need to wait for a quality day or something like that.

**Pull request merged**